consensus/state.go

```diff
@@ -942,6 +946,8 @@ func (cs *State) handleTimeout(ti timeoutInfo, rs cstypes.RoundState) {
 
 		cs.enterPrevote(ti.Height, ti.Round)
 
+	case cstypes.RoundStepDelaydPrevote:
+		cs.enterPrevote(ti.Height, ti.Round)
```

Comment:
Shouldn't you call `doPrevote`here?
If you call `enterPrevote` here again, the `defer` in method `enterPrevote` will be called twice. Not catastrophic, but there'll be a few duplicated events that may confuse external observers (e.g., relayers)

Reply:
Thanks @sergio-mena for your comment,  you are right, the duplicate events needs to be considered carefully which is not yet the case in the current prototype implementation. But, will certainly address this in the full version.